benchmarks/Benchmarks.csproj

```diff
@@ -11,7 +11,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <CodeAnalysisRuleSet>$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), 'OpenTelemetry.sln'))/build/OpenTelemetry.prod.loose.ruleset</CodeAnalysisRuleSet>
+    <CodeAnalysisRuleSet>$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), 'OpenTelemetry.sln'))/build/OpenTelemetry.prod.ruleset</CodeAnalysisRuleSet>
```

Comment:
The idea is to drop the `loose` ruleset altogether? If not, sounds like Benchmark could make use of it.

Reply:
I think, in the end, every project will point to prod.ruleset instead of loose.ruleset. Those that doesn't have i will add separately.
@SergeyKanzhelev , is that right?

Reply:
I think we should just have the single ruleset to be consistent.

Reply:
there is one for tests. Mostly to not require documentation on the test methods.

Reply:
Should we maintain one or use the loose for tests? Let me know I can change it :)

Reply:
there is a separate file for tests. I hope we can remove `loose` one completely and only have `prod` and `test` to choose from

Reply:
@SergeyKanzhelev , benchmark should be test as well? and samples?

Reply:
I agree, prod and test makes sense the most sense and benchmark, samples & tests projects can use the loser version.

Reply:
ok! thanks! i will update my PR's to use this rule:
- benchmark & samples -> loose
- tests -> tests
- others -> prod

Reply:
@MikeGoldsmith , just updated! Thanks for reviewing! :)